File: backend/src/test/java/org/luxons/sevenwonders/test/TestUtils.java
package org.luxons.sevenwonders.test;

import org.springframework.messaging.Message;
import org.springframework.messaging.MessageChannel;
import org.springframework.messaging.simp.SimpMessagingTemplate;

public class TestUtils {

    public static SimpMessagingTemplate createSimpMessagingTemplate() {
        MessageChannel messageChannel = new MessageChannel() {
            @Override
            public boolean send(Message<?> message) {
                return true;
            }

            @Override
            public boolean send(Message<?> message, long timeout) {
                return true;
            }
        };
        return new SimpMessagingTemplate(messageChannel);
    }
}

File: backend/src/test/kotlin/org/luxons/sevenwonders/test/TestUtils.kt
package org.luxons.sevenwonders.test

import org.springframework.messaging.Message
import org.springframework.messaging.MessageChannel
import org.springframework.messaging.simp.SimpMessagingTemplate

object TestUtils {

    fun createSimpMessagingTemplate(): SimpMessagingTemplate {
        val messageChannel = object : MessageChannel {
            override fun send(message: Message<*>): Boolean {
                return true
            }

            override fun send(message: Message<*>, timeout: Long): Boolean {
                return true
            }
        }
        return SimpMessagingTemplate(messageChannel)
    }
}
